### PR TITLE
added crawler to find flyway packages in maven central

### PR DESCRIPTION
**Pull request by prospero238**
Reports the various versions of flyway that are available in maven central.
Intended to be used by the flyway runner plugin in an upcoming release.

### CODE REVIEW

File: flyway.groovy [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  return wc
+}
+
+def listFromMavenRepo() {
+  def json = []
+  def url = "https://repo1.maven.org/maven2/org/flywaydb/flyway-commandline/";
+
+  def wc=webclient();
+  HtmlPage p = wc.getPage(url);
+  def pattern = Pattern.compile("^([0-9][0-9\\.]+)/\$");
```
Comment (daniel-beck) [inline flyway.groovy]: Seems a bit restrictive.

File: flyway.groovy [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      json.addAll( os_versions(url + ver, ver, wc))
+    }
+  }
+  return json
+}
+
+def os_versions(url, version, wc) {
+  json=[]
+  HtmlPage page= wc.getPage(url)
+  def pattern = Pattern.compile("flyway-commandline-" + ver + "(\\.tar\\.gz|-([a-z]+)-x64(\\.zip|\\.tar.gz))\$")
```
Comment (daniel-beck) [inline flyway.groovy]: This as well.

Comment (prospero238) [inline flyway.groovy]: It is indeed restrictive to those packages that the accompanying installer knows how to handle.  The pattern is an attempt to gather the 4 available platform assemblies of each flyway version (and make it easy to extract each of those assemblies's name, i.e. windows, linux, etc).

I'm open to suggestions on how to do this better.  For example, I could parse each version's pom to divine what assemblies exist.

File: flyway.groovy (hunk 1 above)
Comment (prospero238) [inline flyway.groovy]: Fair enough, I've committed a change that merely looks for links that begin with a number.

**Comment by prospero238** [comment]
If this PR is approved,  may I request that it be merged?

**Comment by pskumar448** [comment]
@daniel-beck 
Is there anything to be updated before merging or Is it fine to merge?
Let us know we are waiting for this and based on this we have to plan for release of the plugin.

Thanks.

**Pull request merged**